From 3adf4593132bb7f06ecbb5fe04ff3125c6b80f23 Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Wed, 1 Nov 2017 17:56:23 +0530
Subject: [PATCH 060/767] iommu/of: appropriately get the coherency for fsl-mc
 device

The coherency of fsl-mc devices is provided by MC and is not
governed by device tree. This patch gets the coherency of the
correctly for the fsl-mc devices

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/of/device.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/of/device.c b/drivers/of/device.c
index 33d85511d790..f5f758308e1a 100644
--- a/drivers/of/device.c
+++ b/drivers/of/device.c
@@ -14,6 +14,7 @@
 
 #include <asm/errno.h>
 #include "of_private.h"
+#include "../staging/fsl-mc/include/mc.h"
 
 /**
  * of_match_device - Tell if a struct device matches an of_device_id list
@@ -152,7 +153,10 @@ int of_dma_configure(struct device *dev, struct device_node *np, bool force_dma)
 	dev->coherent_dma_mask &= mask;
 	*dev->dma_mask &= mask;
 
-	coherent = of_dma_is_coherent(np);
+	if (dev_is_fsl_mc(dev))
+		coherent = fsl_mc_is_dev_coherent(dev);
+	else
+		coherent = of_dma_is_coherent(np);
 	dev_dbg(dev, "device is%sdma coherent\n",
 		coherent ? " " : " not ");
 
-- 
2.17.0

